doc: add custom tracing config example for DB2 (#7778)
OpenLiberty/docs: modules/ROOT/pages/jdbc-tracing.adoc

For more information about valid values for the `traceLevel` attribute, see https://www.ibm.com/support/pages/collecting-data-tracing-ibm-data-server-driver-jdbc-and-sqlj[Table 1. DB2 JDBC trace constants].

[#DB2Custom]
=== DB2 JCC customized tracing

> By default, the DB2 JCC driver tracing is combined with Liberty tracing because the DB2 logwriter provided by Liberty routes trace to the same file. However, Liberty also supports separating DB2 JCC trace from Liberty trace. You can use this function to provide a stand-alone DB2 JCC JDBC trace to DB2 support, if needed.

Reviewer: ```suggestion
By default, DB2 JCC driver tracing is combined with Liberty tracing because the Liberty-provided DB2 logwriter directs trace output to the same file. However, Liberty also offers the option to separate DB2 JCC trace from Liberty trace. You can use this option to provide a stand-alone DB2 JCC JDBC trace to DB2 support, if needed.
```

> This cusomized tracing can be set on a per DataSource basis, such as the example below:

Reviewer: ```suggestion
You can set this customized tracing for each configured DataSource in your `server.xml` file, as shown in the following example:
```

[source, xml]
----
<logging traceSpecification="*=info:RRA=all" />

<datasource ... >
  <properties.db2.jcc
    traceLevel="-1" traceDirectory="${server.output.dir}/logs/" traceFile="jcc.trace"
    traceFileAppend="true" traceOption="0"/>
</datasource>
----

> 

Reviewer: ```suggestion
The following attributes customize the configuration of the DB2 driver trace output:
```

- `traceLevel` : This property specifies the level of trace.

> - `traceDirectory` : This property specifies the directory to into which trace is written.

Reviewer: ```suggestion
- `traceDirectory` : This property specifies the directory where trace is written.
```

> - `traceFile` : This property specifies the name of the file into which trace is written.

Reviewer: ```suggestion
- `traceFile` : This property specifies the name of the trace output file.
```

> - `traceFileAppend` : This property specifies wheather to append to (true) or overwrite (false) the `traceFile`.

Reviewer: ```suggestion
- `traceFileAppend` : This property specifies whether to append to (`true`) or overwrite (`false`) the trace output to the specified file.
```

> - `traceOption` (Optional) : This property when set to `0` specifies that a single trace file will be generated, and that no limit to the size of the file exists. 

Reviewer: ```suggestion
- `traceOption` (Optional) : When you set this property  to `0`,  a single trace file with no size limit is generated. 
```

Reviewer: this property is marked optional- are the others all required?

what is the default file size/ rollover if you dont set it?

> For additional information about these properties and additional configuration settings, such as `traceFileSize` and `traceFileCount`, see https://www.ibm.com/docs/en/db2/11.5?topic=information-data-server-driver-jdbc-sqlj-configuration-properties(IBM Data Server Driver for JDBC and SQLJ configuration properties

Reviewer: ```suggestion
For more information about these properties and other configuration settings, such as `traceFileSize` and `traceFileCount`, see link:https://www.ibm.com/docs/en/db2/11.5?topic=information-data-server-driver-jdbc-sqlj-configuration-properties[IBM Data Server Driver for JDBC and SQLJ configuration properties].
```

Reviewer: is the information at the link DB2 version-specific? If not, we should use https://www.ibm.com/docs/en/db2/latest?topic=information-data-server-driver-jdbc-sqlj-configuration-properties for the URL so that the link does not go out of date. From what Im seeing, 11.5 is no longer the latest version.

You can set this customized tracing for each configured DataSource in your `server.xml` file, as shown in the following example:

[source, xml]
----
<logging traceSpecification="*=info:RRA=all" />

<datasource ... >
  <properties.db2.jcc
    traceLevel="-1" traceDirectory="${server.output.dir}/logs/" traceFile="jcc.trace"
    traceFileAppend="true" traceOption="0"/>
</datasource>
----

The following attributes customize the configuration of the DB2 driver trace output:

- `traceLevel` (required): This property specifies the level of trace.
- `traceDirectory` (required): This property specifies the directory where trace is written.
- `traceFile` (required): This property specifies the name of the trace output file.
- `traceFileAppend` (optional): This property specifies whether to append to (`true`) or overwrite (`false`) the trace output to the specified file. Default value is `false`.

> - `traceOption` (optional) : When set to `0` the `traceFileSize` and `traceFileCount` properties are ignored, and a single trace file with no size limit is generated. To enable circular tracing set this property to `1`. Default value is `0`.

Reviewer: ```suggestion
- `traceOption` (optional) : When this property is set to `0`, the `traceFileSize` and `traceFileCount` properties are ignored, and a single trace file with no size limit is generated. To enable circular tracing, set this property to `1`. Default value is `0`.
```
